apache-pulsar/src/main/java/io/debezium/examples/apache/pulsar/PulsarProducer.java

```diff
@@ -0,0 +1,171 @@
+        propConfig = new Properties();
+        try {
+            propConfig.load(propsInputStream);
+        }
+        catch (IOException e) {
+            logger.error("Couldn't load properties", e);
+        }
+
+        PropertyLoader.loadEnvironmentValues(propConfig);
+        config = Configuration.from(propConfig);
+
+        keyConverter = new JsonConverter();
+        keyConverter.configure(config.asMap(), true);
+        valueConverter = new JsonConverter();
+        valueConverter.configure(config.asMap(), false);
+    }
+
+    private Producer<String> getProducer(String topic) {
+        String topicFormat = propConfig.getProperty("pulsar.topic");
+        String topicURI = MessageFormat.format(topicFormat, topic);
+        Producer<String> producer = producerHashMap.get(topic);
+
+        if (producer == null) {
+            try {
+                producer = client.newProducer(Schema.STRING)
+                        .topic(topicURI)
+                        .create();
+            }
+            catch (PulsarClientException e) {
+                throw new RuntimeException(e);
+            }
+
+            producerHashMap.put(topic, producer);
+        }
+
+        return producer;
+    }
+
+    @Override
+    public void run() {
+        final EmbeddedEngine engine = EmbeddedEngine.create()
+                .using(config)
+                .using(this.getClass().getClassLoader())
+                .using(Clock.SYSTEM)
+                .notifying(this::sendRecord)
+                .build();
+
+        try {
+            client = PulsarClient.builder()
+                    .serviceUrl(propConfig.getProperty("pulsar.broker.address"))
+                    .build();
+        }
+        catch (PulsarClientException e) {
+            throw new RuntimeException(e);
+        }
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(engine);
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            logger.info("Requesting embedded engine to shut down");
+            try {
+                engine.stop();
+
+                producerHashMap.forEach((topic, producer) -> {
+                    logger.info(String.format("Closing producer for topic %s", topic));
+                    try {
+                        producerHashMap.get(topic).close();
+                    }
+                    catch (PulsarClientException e) {
+                        logger.error("Couldn't close producer", e);
+                    }
+                    logger.info("Producer Closed");
+
+                });
+                client.close();
+            }
+            catch (PulsarClientException e) {
+                throw new RuntimeException(e);
+            }
+        }));
+
+        awaitTermination(executor);
+    }
+
+    /**
+     * For Every record this method will be invoked* @param record
```

Comment:
Capital `E`, `@param` should go on its own line

Reply:
Fixed.